File: bindings/pygegl/tests/op_reference.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#This is a Python version of the op_reference.rb sample script written
#by Øyvind Kolås for the Ruby binding (rgegl) of gegl.

import gegl

operations = gegl.gegl_list_operations()

print "<html><body>"

print "<ul>"
for operation_name in operations:
   print "<li><a href='%s'>%s</li>", operation_name, operation_name

print "</ul>"

for operation_name in operations:
    print "<a name='%s'></a><h3>%s</h3>", operation_name, operation_name
    print "<table>"
    Gegl.properties(operation_name).each {|property|
        case property.gtype
        when GLib::Type["GParamDouble"]
                print "<tr><td>#{property.value_type}</td><td>#{property.name}</td><td>#{property.blurb}</td><td>min:#{property.minimum} max:#{property.maximum} default:#{property.default}</td></tr>"
        else
                print "<tr><td>#{property.value_type}</td><td>#{property.name}</td><td>#{property.blurb}</td></tr>"
        end

    }
    print "</table>"
}

print "</body></html>"

File: bindings/rgegl/samples/op-reference.rb
#!/usr/bin/env ruby

$LOAD_PATH.unshift "../src"
$LOAD_PATH.unshift "../src/lib"

require 'glib2'
require 'gegl'


puts "<html><body>"

puts "<ul>"
Gegl.operations.each {|operation_name| puts "<li><a href='##{operation_name}'>#{operation_name}</li>"}
puts "</ul>"

Gegl.operations.each {|operation_name|
    puts "<a name='#{operation_name}'></a><h3>#{operation_name}</h3>"
    puts "<table>"
    Gegl.properties(operation_name).each {|property|
        case property.gtype
        when GLib::Type["GParamDouble"]
                puts "<tr><td>#{property.value_type}</td><td>#{property.name}</td><td>#{property.blurb}</td><td>min:#{property.minimum} max:#{property.maximum} default:#{property.default}</td></tr>"
        else
                puts "<tr><td>#{property.value_type}</td><td>#{property.name}</td><td>#{property.blurb}</td></tr>"
        end

    }
    puts "</table>"
}

puts "</body></html>"
